hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -74,8 +75,13 @@ public List<String> getAllPartitionPaths() throws IOException {
       return FSUtils.getAllPartitionFoldersThreeLevelsDown(fs, datasetBasePath);
     }
 
+    return getPartitionPathsWithPrefix("");
+  }
+
+  @Override
+  public List<String> getPartitionPathsWithPrefix(String prefix) throws IOException {
     List<Path> pathsToList = new CopyOnWriteArrayList<>();
-    pathsToList.add(basePath);
+    pathsToList.add(StringUtils.isNullOrEmpty(prefix) ? new Path(datasetBasePath) : new Path(datasetBasePath, prefix));
```

Comment:
This implementation currently assumes that either a) prefix is "" or b) prefix is an actual relative path (instead of, say, prefix of the name of the folder) which could be misleading

Instead, i'd suggest we do the following:

 - Let's create a separate method `getPartitionPathsWithPrefix(Option<String> prefix)`
 - Not allow empty prefixes to be passed in
 - Make sure that implementation can handle actual prefixes (instead of currently digesting only the full-paths)

Reply:
Actually, it could be a prefix of the name of the folder, e.g., prefix=`year=1992`, and the actual structure is `year=1992/month=xx/day=xx`.

The reason I added a check here (i.e., Empty check) is that the Path constructor cannot handle empty "" input for its second parameter.